gcp-jobs/involuntary-dissolutions/src/involuntary_dissolutions/config.py

```diff
@@ -0,0 +1,114 @@
+"""All of the configuration for the service is captured here."""
+
+import os
+
+from dotenv import find_dotenv, load_dotenv
+
+load_dotenv(find_dotenv())
+
+CONFIGURATION = {
+    "development": ".config.DevConfig",
+    "testing": ".config.TestConfig",
+    "production": ".config.ProdConfig",
+    "default": ".config.ProdConfig"
+}
+
+
+def get_named_config(config_name: str = "production"):
+    """Return the configuration object based on the name."""
+    if config_name in ["production", "staging", "default"]:
+        config = ProdConfig()
+    elif config_name == "testing":
+        config = TestConfig()
+    elif config_name == "development":
+        config = DevConfig()
+    else:
+        raise KeyError(f"Unknown configuration '{config_name}'")
+    return config
+
+
+class _Config:  # pylint: disable=too-few-public-methods
+    """Base class configuration."""
+
+    SERVICE_NAME = "dissolutions-job"
+    PROJECT_ROOT = os.path.abspath(os.path.dirname(__file__))
+
+    LD_SDK_KEY = os.getenv("LD_SDK_KEY", None)
+
+    SERVERS = os.getenv("SERVERS", None)
+    CLUSTER_ID = os.getenv("CLUSTER_ID", None)
+    CLIENT_NAME = os.getenv("CLIENT_NAME", None)
+    FILER_SUBJECT = os.getenv("FILER_SUBJECT", "entity.filer")
+    ENTITY_EVENTS_SUBJECT = os.getenv("ENTITY_EVENTS_SUBJECT", "entity.events")
+
+    AUTH_SVC_URL = os.getenv("AUTH_URL", "http://")
+    ACCOUNT_SVC_AUTH_URL = os.getenv("ACCOUNT_SVC_AUTH_URL")
+    ACCOUNT_SVC_CLIENT_ID = os.getenv("ACCOUNT_SVC_CLIENT_ID")
+    ACCOUNT_SVC_CLIENT_SECRET = os.getenv("ACCOUNT_SVC_CLIENT_SECRET")
+
+    SECRET_KEY = "a secret"
+
+    SQLALCHEMY_TRACK_MODIFICATIONS = False
+
+    ALEMBIC_INI = "migrations/alembic.ini"
+
+    # POSTGRESQL
+    DB_USER = os.getenv("DATABASE_USERNAME", "")
+    DB_PASSWORD = os.getenv("DATABASE_PASSWORD", "")
+    DB_NAME = os.getenv("DATABASE_NAME", "")
+    DB_HOST = os.getenv("DATABASE_HOST", "")
+    DB_PORT = os.getenv("DATABASE_PORT", "5432")
+    SQLALCHEMY_DATABASE_URI = f"postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{int(DB_PORT)}/{DB_NAME}"
+
+    TESTING = False
+    DEBUG = False
+
+    STAGE_1_DELAY = int(os.getenv("STAGE_1_DELAY", "42"))
+    STAGE_2_DELAY = int(os.getenv("STAGE_2_DELAY", "30"))
+
+
+class DevConfig(_Config):  # pylint: disable=too-few-public-methods
+    """Development environment configuration."""
+
+    TESTING = False
+    DEBUG = True
+
+
+class TestConfig(_Config):  # pylint: disable=too-few-public-methods
+    """In support of testing only used by the py.test suite."""
+
+    DEBUG = True
+    TESTING = True
+
+    # POSTGRESQL
+    DB_USER = os.getenv("DATABASE_TEST_USERNAME", "")
+    DB_PASSWORD = os.getenv("DATABASE_TEST_PASSWORD", "")
+    DB_NAME = os.getenv("DATABASE_TEST_NAME", "")
+    DB_HOST = os.getenv("DATABASE_TEST_HOST", "")
+    DB_PORT = os.getenv("DATABASE_TEST_PORT", "5432")
+    SQLALCHEMY_DATABASE_URI = f"postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{int(DB_PORT)}/{DB_NAME}"
+
```

Comment:
and can you add `GCP_AUTH_KEY = None` here pls

Reply:
done